### ISSUE TITLE
False positive from `type_repetition_in_bounds`

### ISSUE THREAD

**Issue by ethanboxx** [rust-lang/rust-clippy#5839]
I tried this code:

```rust
use std::collections::HashMap;

trait Example {}

struct Map<K, V>(HashMap<K, V>);

impl<K, V> Map<K, V>
where
    HashMap<K, V>: Clone,
    HashMap<String, V>: Example,
{
}

```
I get the warning

```rust
warning: this type has already been used as a bound predicate
  --> src/main.rs:14:5
   |
14 |     HashMap<String, V>: Example,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `-W clippy::type-repetition-in-bounds` implied by `-W clippy::pedantic`
   = help: consider combining the bounds: `HashMap<String, V>: Clone + Example`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#type_repetition_in_bounds
```

`HashMap<K, V>` and `HashMap<String, V>` are different types so this lint should not show.

### Meta

- `cargo clippy -V`: clippy 0.0.212 (346aec9 2020-07-11)
- `rustc -Vv`:
```
rustc 1.46.0-nightly (346aec9b0 2020-07-11)
binary: rustc
commit-hash: 346aec9b02f3c74f3fce97fd6bda24709d220e49
commit-date: 2020-07-11
host: x86_64-unknown-linux-gnu
release: 1.46.0-nightly
LLVM version: 10.0
```

**Comment by ThibsG** [comment]
[Playgrounding](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=de6a62692a15e3f5a0bf8aaecc558a74) this, I cannot reproduce it (Clippy's version 07/24)

As #5761 has been merged recently, maybe it has already been fixed ?

**Comment by ethanboxx** [comment]
Very possible thank you. I will reopen this issue if I run into this problem in future after updating with rustup when a version is avalable with all the components I use.
